options/options_test.cc

```diff
@@ -585,6 +585,22 @@ TEST_F(OptionsTest, GetColumnFamilyOptionsFromStringTest) {
   ASSERT_TRUE(new_cf_opt.memtable_factory != nullptr);
   ASSERT_EQ(std::string(new_cf_opt.memtable_factory->Name()), "SkipListFactory");
   ASSERT_TRUE(new_cf_opt.memtable_factory->IsInstanceOf("SkipListFactory"));
+
+  // blob cache
+  ASSERT_OK(GetColumnFamilyOptionsFromString(
+      config_options, base_cf_opt,
+      "blob_cache={capacity=1M;num_shard_bits=4;"
+      "strict_capacity_limit=true;high_pri_pool_ratio=0.5;};",
+      &new_cf_opt));
+  ASSERT_TRUE(new_cf_opt.blob_cache != nullptr);
+  ASSERT_EQ(new_cf_opt.blob_cache->GetCapacity(), 1024UL * 1024UL);
+  ASSERT_EQ(std::dynamic_pointer_cast<ShardedCache>(new_cf_opt.blob_cache)
+                ->GetNumShardBits(),
+            4);
+  ASSERT_EQ(new_cf_opt.blob_cache->HasStrictCapacityLimit(), true);
+  ASSERT_EQ(std::dynamic_pointer_cast<LRUCache>(new_cf_opt.blob_cache)
+                ->GetHighPriPoolRatio(),
+            0.5);
```

Comment:
Couple of things here (and below)... doing an `std::dynamic_pointer_cast` which results in an empty pointer if the cast fails but then unconditionally dereferencing the pointer kind of defeats the purpose of the checked cast. Also, we could use `ASSERT_NE` here as well to check for nullptr like @riversand963 suggested. So how about something like

```
auto blob_cache = std::dynamic_pointer_cast<ShardedCache>(new_cf_opt.blob_cache);
ASSERT_NE(blob_cache, nullptr);
// further checks for shard bits etc. here using the blob_cache ptr above
```

Reply:
I think it can't be an empty pointer since we initialized the blob cache here. :)   Anyway, I can modify it to make it more defensive.

Reply:
I realize that, it's just that unconditionally dereferencing the result of a dynamic cast is kind of fishy :) BTW, a static cast would also work because we know for a fact it's an LRU cache.

Reply:
Done, let me know if there is any issue.